src/transformers/generation_tf_utils.py

```diff
@@ -547,6 +555,38 @@ def generate(
             input_ids=input_ids, max_length=100, do_sample=True, bad_words_ids=bad_words_ids
         )  # generate sequences without allowing bad_words to be generated
         ```"""
+        num_beams = num_beams if num_beams is not None else self.config.num_beams
```

Comment:
As one can see the original `generate()` method is not really touched in this PR. Here all that is done is to force to use the new `_generate()` method for the greedy search use case. IMO, this can be left as is for now and makes the refactor much easier to split into multiple steps while being sure nothing is broken down the line.